### ISSUE TITLE
i18n: decimal comma separator

### ISSUE THREAD

**Issue by jgpATs2w** [phetsims/dot#68]
Unless there is other repo/mechanism to handle with internationalization, i would suggest to include dot in babel, with a parameter to allow translators to select the decimal separator, and make proper replacement in Utils.toFixed().
As always..., thanks for this great simulations!!

**Comment by jonathanolson** [comment]
@ariel-phet, have we looked into using built-in browser support for number i18n? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat now looks like it meets our minimum browser requirements, and would handle some cases where we would want to change how numbers are displayed:

```js
var number = 123456.789;

// German uses comma as decimal separator and period for thousands
console.log(new Intl.NumberFormat('de-DE').format(number));
// → 123.456,789
```

Doesn't look like a catch-all, as I believe its behavior for Arabic may not be what we've been told is the best:
```js
new Intl.NumberFormat( 'ar' ).format( 123456.789 ) // "١٢٣٬٤٥٦٫٧٨٩"
```

**Comment by ariel-phet** [comment]
@jonathanolson I don't believe we have looked into built in browser support. I believe we usually have the option to set the separator on a sim by sim basis but perhaps I am mistaken.

Marking for dev-meeting

**Comment by pixelzoom** [comment]
+1 for investigating built-in (browser) support for numbers.

**Comment by samreid** [comment]
> +1 for investigating built-in (browser) support for numbers.

agreed

**Comment by pixelzoom** [comment]
Perhaps we should ask the translator in Rosetta to enter:

decimalSeparator (default to '.')
thousandsSeparator (default to none)

**Comment by samreid** [comment]
Would these strings be in Joist?

**Comment by ariel-phet** [comment]
@oliver-phet will poll some of our translators about this issue

**Comment by pixelzoom** [comment]
@oliver-phet Any reply from translators?

**Comment by oliver-phet** [comment]
The general consensus from trusted translators is that this is not a priority. A couple said it would help to have the control, others didn't mind.

**Comment by ariel-phet** [comment]
We will plan to make this automatic for each locale with established conventions, work to be done with Rosetta etc work

**Comment by samreid** [comment]
@kathy-phet said:

> [I'm] wondering about the challenges around displaying numbers more in the locale of the countries. Has the development team considered this (new?) function https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat. I know its complicated to figure out how to support different number formats - especially with the precision with which we tend to do alignments or other display characteristics. I am just wondering if this function is new and if it is at all helpful for PhET. e.g. if we wanted to display 50.5 as 50,5 when using the German translation.

**Comment by samreid** [comment]
It appears Intl.NumberFormat was introduced around 2013.  We discussed it in this issue https://github.com/phetsims/dot/issues/68 in 2017 and there is a related issue for i18n of the keypad in https://github.com/phetsims/scenery-phet/issues/333.  I’ll post your comment in dot#68 and check the status.

@jbphet or @ariel-phet can you please comment on the status?

**Comment by jbphet** [comment]
This is just one of those features that has never bubbled to the top of the priority list.  @oliver-phet said above:

> The general consensus from trusted translators is that this is not a priority.

We can reconsider and make it a priority if needed.

**Comment by ariel-phet** [comment]
I will try to bring this question up at status meeting this week.

**Comment by samreid** [comment]
new Intl.NumberFormat relies on BCP 47 language tag, so we would likely want to build and maintain a mapping from our codes to the BCP47.

**Comment by oliver-phet** [comment]
https://tools.ietf.org/html/bcp47#section-2

**Comment by jbphet** [comment]
Marking again for dev meeting for two reasons:

1. This is never getting any time spent on it because it's marked as medium priority, and there are always higher priority things happening.  Should we schedule it to become high priority at some point?
2. We got a request for this feature from a translator, see https://github.com/phetsims/rosetta/issues/252
3. Should this issue be moved?  The dot repo doesn't seem like the most obvious place for it.

**Comment by jbphet** [comment]
The dev team discussed this in the 1/25/2021 supplemental developer meeting, and I will follow up by first having @oliver-phet survey the translators in an effort to gauge how important and valuable this feature would be to them, and then to work with @kathy-phet to decide whether to allocate time to this now or defer.

**Comment by samreid** [comment]
My note from https://github.com/phetsims/rosetta/issues/252#issuecomment-766972319

> In my opinion, PhET should look "at home" in other languages, and the only reason to delay or forego this issue is if it is prohibitively expensive.

**Comment by pixelzoom** [comment]
Elaborating on the survery idea...  The questions that I would ask are something like, for your locale:

* What decimal separator is used?
* Are students used to seeing the English separator (dot) and does it confuse them?
* Is using the correct separator essential, nice to have, or not really necessary?
* Is usabity or learning compromised if we do nothing, and continue to use the English separator?

**Comment by oliver-phet** [comment]
Thanks for the ideas @pixelzoom

**Comment by oliver-phet** [comment]
We've [received 32 responses](https://docs.google.com/forms/d/1gSJAC5OJUZc1_22SJcUQc7VmePbn00i-KKtGiosZFLE/edit#responses) to the survey I sent to our full list of trusted translators (150 translators).

It seems this is a priority feature to translators:

- ~90% view using the correct separator as Essential / Nice to have
- 50% feel usability or learning would be compromised without the feature

Also, it seems that the most used separator isn't a comma or decimal, but a space. 
- ~43% of translators use Comma for decimal and space for thousands in their locale.

Full list of separators used:
16 responses: Comma for decimal and space for thousands: `123 456,789`
8 responses: Comma for decimal and period for thousands: `123.456,789`
6 responses: Period for decimal and comma for thousands: `123,456.789`
1 response: Comma for decimal and no separator for thousands: `123456,789`
1 response: Forward slash for decimal and comma for thousands; `123,456/789`

**Comment by jbphet** [comment]
I'm going to move this issue to the rosetta repo.  There really isn't a perfect place for this to go, since localization of the decimal separator is a very cross-cutting feature, and we're not sure at this point if we would do it automatically or based on translation, but having this issue in dot is clearly sub-optimal.
